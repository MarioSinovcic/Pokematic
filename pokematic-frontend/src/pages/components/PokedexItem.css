.PokedexItem{
    width: 272px !important;
    height: 222px !important;
}

.pokemon-name{
    padding-left: 22px;
    padding-top: 20px;

    font-family: "Trebuchet MS", Helvetica, sans-serif;
    font-style: normal;
    font-weight: 800;
    font-size: 25px;
    line-height: 34px;

    color: #FFFFFF;
}