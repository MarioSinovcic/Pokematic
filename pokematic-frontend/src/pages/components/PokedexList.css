.PokedexList{
   margin-top: 100px !important;
   align-content: center;
   margin: 0 auto;
   width: 100%;
}


.grid-container {
   width: 90%;
   margin: auto;
   padding-left: 20px;
   display: flex;
   flex-flow: wrap;
   align-items: center;
}

 .grid-item{
    height: initial !important;
    background-color: lightblue;
    padding: 5px 5px !important;;
    margin: 34px 55px !important;;
    border-radius: 20px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
 }