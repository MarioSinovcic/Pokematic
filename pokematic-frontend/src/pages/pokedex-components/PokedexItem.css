.PokedexItem{
    width: 272px !important;
    height: 222px !important;
    position: relative;
    overflow: hidden;
}

.background-enabler{
    z-index: 2;
    position: relative;
}

.pokeball-background{
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 250px;
    opacity: 0.15;

    background-image: url('../../images/background_pokeball.png');
    background-repeat: no-repeat;
    background-size: 170px;
    background-position-x: 100px;
    background-position-y: 45px;
    background-position: fixed;
}

.pokemon-number{
    width: 90%;
    text-align: right;
    margin-top: 11px;
    margin-left: 15px;

    font-family: "Trebuchet MS", Helvetica, sans-serif;
    font-style: normal;
    font-weight: 800;
    font-size: 20px;
    line-height: 27px;

    color: #58BFEB;
}

.pokemon-name{
    position: static;
    margin-left: 22px;
    margin-top: -20px;

    font-family: "Trebuchet MS", Helvetica, sans-serif;
    font-style: normal;
    font-weight: 900;
    font-size: 28px;
    line-height: 34px;
    color: whitesmoke;
}
    
.pokemon-image-wrapper{
    width: 150px;
    height: 150px;
    float: right;
}

.pokemon-image{
    height: 120%;
    margin: -10px -35px !important;
}

.pokemon-type{
    margin-left: 22px;
    margin-top: 130px;
    width: 64px;
    text-align: center;

    padding: 7px 7px;
    font-family: "Trebuchet MS", Helvetica, sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 10px;
    color: #FFFFFF;

    border-radius: 15px;
    background-color: #44B8F9;
}