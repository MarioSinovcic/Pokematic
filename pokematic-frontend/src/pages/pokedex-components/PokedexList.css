.PokedexList{
   margin-top: 100px !important;
   align-content: center;
   margin: 0 auto;
   width: 100%;
   height: 100%;
}

.grid-container {
   width: 100%;
   height: 100%;
   display: flex;
   flex-flow: wrap;
   align-items: center;
   justify-content: center;
}

 .grid-item{
    height: initial !important;
    background-color: #c4c4c4;
    margin: 34px 45px !important;;
    border-radius: 20px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    transition: 0.6s;
 }

 .grid-item:hover{
   margin-top: 20px !important;
   box-shadow: 0px 9px 4px rgba(0, 0, 0, 0.2);
   transition: 0.3s;
}