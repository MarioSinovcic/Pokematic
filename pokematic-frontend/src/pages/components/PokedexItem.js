import React from 'react';
import './PokedexItem.css'

function PokedexItem() {
  return (
    <div className="PokedexItem">
        <div className="pokemon-name">
          Pokemon Name
        </div>
    </div>
  );
}

export default PokedexItem;
