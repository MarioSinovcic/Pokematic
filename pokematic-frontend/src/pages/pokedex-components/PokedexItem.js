import React from 'react';
import './PokedexItem.css'
import Label from '../../shared-components/Label';
import PokemonMappingUtil from './PokemonMappingUtil';

class PokedexItem extends React.Component {

  render() {
    
    const firstType = this.props.pokemonTypeList && this.props.pokemonTypeList[1];
    const secondType = this.props.pokemonTypeList && this.props.pokemonTypeList[0];

    return (
      <div className="PokedexItem">
        <div className="pokeball-background"></div>
        <div className="background-enabler">
          <div className="pokemon-number">
            #{this.props.pokemonNumber + 1}
          </div>
          <div className="pokemon-name">
            {this.props.pokemonName}
          </div>
          <div className="pokemon-image-wrapper">
            <img className="pokemon-image" src={this.props.pokemonImage} alt=""></img>
          </div>
          <div className="pokemon-type">
         {firstType ?
            <Label labelText={firstType} color={PokemonMappingUtil.mapTypeToLabelColors(firstType)}/>
            : null}
          <Label labelText={secondType} color={PokemonMappingUtil.mapTypeToLabelColors(secondType)}/>
          </div>
        </div>
      </div>
    );
  }

}


export default PokedexItem;
