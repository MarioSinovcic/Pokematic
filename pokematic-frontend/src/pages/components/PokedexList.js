import React from 'react';
import PokedexItem from './PokedexItem'
import './PokedexList.css'

function PokedexList() {

  // This a temporary solution for rendering all of the pokemon cards
  var fakePokemon = [];
  for (var i = 0; i < 150; i++) {
    fakePokemon.push(i)
  }

  var pokemonToRender = fakePokemon.map(item => <div class="grid-item"><PokedexItem/></div>)

  return (
    <div className="PokedexList">
      <div className="pokeball-background-black"></div>
      <div className="background-enabler">
        <div class="grid-container">
          {pokemonToRender}
        </div>
      </div>
    </div>
  );
}

export default PokedexList;
