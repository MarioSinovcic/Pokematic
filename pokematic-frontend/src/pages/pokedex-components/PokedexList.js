import React from 'react';
import PokedexItem from './PokedexItem'
import './PokedexList.css'
import { connect } from 'react-redux';
import { addPokemonNames, addPokemonTypes, togglePokemonLoad, addPokemonData } from '../../actions/actions'
import PokedexMappingUtil from './PokemonMappingUtil';

class PokedexList extends React.Component {

  state = {
    pokemon: this.props.pokemonData || [],
  }

  componentWillMount() {
    if (!this.props.isLoaded) {
      this.fetchPokemonData();
      this.props.togglePokemonHasLoaded();
    }
  }


  async fetchPokemonData() {

    // Get a list of pokemon Names and their URLs
    await fetch("https://pokeapi.co/api/v2/pokemon/?offset=0&limit=151")
      .then(response => response.json())
      .then(data => {
        this.props.addPokemon(data.results);
      })

    await this.props.pokemonMap.map((pokemon, i) => {
      return (
        this.fetchPokemonTypes(pokemon.url)
      )
    })

    // Add data to Redux Store
    this.props.addPokemonData(this.populatePokemon());


    this.setState({
      pokemon: this.populatePokemon()
    })
  }

  async fetchPokemonTypes(pokemonURL) {
    await fetch(pokemonURL)
      .then(response => response.json())
      .then(data => {
        const name = data.name;
        const types = [data.types[0].type.name, (data.types[1] && data.types[1].type.name)];
        this.props.addPokemonTypes(name, types);
      })
  }

  populatePokemon() {

    var populatedPokemon = []

    this.props.pokemonMap.map((pokemonData, i) => {
      return (
        populatedPokemon.push(
          [i,
            pokemonData && pokemonData.name,
            "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/" + (i + 1) + ".png",
          ])
      );
    })
    return populatedPokemon;
  }
  

  getPokemonTypes(pokemonTypesList, pokemonName) {
    const typesArray = pokemonTypesList.find(pokemon =>
      pokemonName === pokemon[0]
    )

    if (typesArray) {
      // Return an array of just the types without the pokemon name
      return typesArray[1];
    }
  }



  render() {
    return (
      <div className="PokedexList">
        <div className="grid-container">
          {this.state.pokemon.map((pokemonData, i) => {

            const pokemonName = pokemonData[1];
            const pokemonTypeList = this.getPokemonTypes(this.props.pokemonTypes, pokemonName);
            
            return (


              <div key={i} className="grid-item" style={{
                backgroundColor: PokedexMappingUtil.mapTypeToBackgroundColors(
                  pokemonTypeList ?
                  // Take the color of second type if it exists because it seems to usually be the main type of the pokemon
                    pokemonTypeList[1] ?
                      pokemonTypeList[1]
                      : pokemonTypeList[0]
                    : ""

                )
              }}>
                <PokedexItem
                  pokemonNumber={pokemonData[0]}
                  pokemonName={pokemonName}
                  pokemonImage={pokemonData[2]}
                  pokemonTypeList={pokemonTypeList} />
              </div>)
          })}

        </div>
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    pokemonMap: state.pokemonInfo,
    pokemonTypes: state.pokemonTypes,
    pokemonData: state.pokemonData,
    isLoaded: state.isLoaded,
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    togglePokemonHasLoaded: () => {
      dispatch(togglePokemonLoad())
    },
    addPokemonData: (pokemon) => {
      dispatch(addPokemonData(pokemon))
    },
    addPokemon: (pokemon) => {
      dispatch(addPokemonNames(pokemon))
    },
    addPokemonTypes: (pokemonName, pokemonType) => {
      dispatch(addPokemonTypes(pokemonName, pokemonType))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PokedexList);
