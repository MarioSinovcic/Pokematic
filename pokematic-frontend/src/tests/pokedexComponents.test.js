import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import rootReducers from '../redux/reducers';
import PokedexItem from '../pages/pokedex-components/PokedexItem';
import PokedexList from '../pages/pokedex-components/PokedexList';

describe('pokedex components', () => {

    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })

    describe('snapshot matching', () => {

        it('Pokedex Item matches snapshot' , () => {
            const pokeItem = renderer.create(<Provider store={store}>
                <PokedexItem />
            </Provider>).toJSON();
            expect(pokeItem).toMatchSnapshot();
        });

        it('Pokedex List matches snapshot' , () => {
            const dummyPokemonList = [];

            const pokeList = renderer.create(<Provider store={store}>
                <PokedexList 
                pokemonCollection={dummyPokemonList}/>
            </Provider>).toJSON();
            expect(pokeList).toMatchSnapshot();
        });
    })

});