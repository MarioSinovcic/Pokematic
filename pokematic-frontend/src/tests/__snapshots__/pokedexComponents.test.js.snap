// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokedex components snapshot matching Pokedex Item matches snapshot 1`] = `
<div
  className="PokedexItem"
>
  <div
    className="pokeball-background"
  />
  <div
    className="background-enabler"
  >
    <div
      className="pokemon-number"
    >
      #
      NaN
    </div>
    <div
      className="pokemon-name"
    />
    <div
      className="pokemon-image-wrapper"
    >
      <img
        alt=""
        className="pokemon-image"
      />
    </div>
    <div
      className="pokemon-type"
    >
      <div
        id="Label"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text team-label pokemon-label"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": null,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`pokedex components snapshot matching Pokedex List matches snapshot 1`] = `
<div
  className="PokedexList"
>
  <div
    className="grid-container"
  />
</div>
`;
